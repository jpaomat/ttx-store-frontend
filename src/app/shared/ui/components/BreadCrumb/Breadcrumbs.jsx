import { Link } from 'react-router-dom';
import useBreadcrumbs from 'use-react-router-breadcrumbs';
import PropTypes from 'prop-types';
import './Breadcrumbs.scss';

export const Breadcrumbs = ({ routes }) => {
	const breadcrumbs = useBreadcrumbs(routes, { disableDefaults: true });

	return (
		<div className='breadcrumbs'>
			{breadcrumbs.map(({ breadcrumb, match }, index) => (
				<div className='bc breadcrumbs__item' key={match.pathname}>
					<Link className={index} to={match.pathname || ''}>
						{breadcrumb}
					</Link>
					{index < breadcrumbs.length - 1 && '>'}
				</div>
			))}
		</div>
	);
};

Breadcrumbs.defaultProps = {
	routes: [{}],
};

Breadcrumbs.propTypes = {
	routes: PropTypes.arrayOf(PropTypes.shape({})),
};
