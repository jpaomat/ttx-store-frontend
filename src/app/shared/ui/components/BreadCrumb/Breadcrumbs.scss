.breadcrumbs,
.bc,
.content {
	display: flex;
}

.breadcrumbs {
	display: flex;
	align-items: center;
	&__item {
		align-items: center;
		white-space: nowrap;
		overflow: hidden;
	}
}

.breadcrumbs span {
	display: flex;
	align-items: center;
}
