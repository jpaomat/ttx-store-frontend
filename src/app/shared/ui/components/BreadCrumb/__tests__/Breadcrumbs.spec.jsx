import { render } from '@testing-library/react';
import { Breadcrumbs } from '../Breadcrumbs';
import { MemoryRouter } from 'react-router-dom';
import { defaultMock } from './mocks';

// jest.mock('use-react-router-breadcrumbs', () => ({
//     // ...jest.requireActual('use-react-router-breadcrumbs'),
//     // useBreadcrumbs : () => (
//     //     [
//     //         {
//     //             breadcrumb: {
//     //                 children : {
//     //                     children: 'Productos'
//     //                 }
//     //             },
//     //             key: '/product',
//     //             match: {
//     //                 pathname: 'Producto'
//     //             }
//     //         }
//     //     ]
//     // ),
// }));

describe('<Breadcrumbs /> component', () => {
	test('should render component', () => {
		const { container } = render(
			<MemoryRouter>
				<Breadcrumbs {...defaultMock} />
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});
});
