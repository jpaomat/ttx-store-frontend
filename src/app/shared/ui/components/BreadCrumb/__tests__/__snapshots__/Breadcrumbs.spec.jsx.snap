// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumbs /> component should render component 1`] = `
<div>
  <div
    class="breadcrumbs"
  />
</div>
`;
